File: backend/internal/dto/auth_dto.go
package dto

type SendEmailVerificationRequest struct {
	Email string `json:"email" binding:"required,email"`
}

type VerifyEmailCodeRequest struct {
	Email string `json:"email" binding:"required,email"`
	OTP   string `json:"otp" binding:"required,len=6"`
}

type CompleteRegistrationRequest struct {
	VerificationToken string `json:"verification_token" binding:"required"`
	FullName          string `json:"full_name" binding:"required,min=3"`
	StudentCode       string `json:"student_code"` // Optional - empty for lecturers
	Password          string `json:"password" binding:"required,min=6"`
}

type ResendOTPRequest struct {
	Email string `json:"email" binding:"required,email"`
}

// Login
type UserLoginRequest struct {
	Identifier string `json:"identifier" binding:"required"`
	Password   string `json:"password" binding:"required"`
}

type CompleteGoogleSetupRequest struct {
	SetupToken  string `json:"setup_token" binding:"required"`
	FullName    string `json:"full_name" binding:"required,min=3"`
	StudentCode string `json:"student_code"` // Optional - empty for lecturers
}
type RefreshRequest struct {
	RefreshToken string `json:"refresh_token" binding:"required"`
}

type LogoutRequest struct {
	AccessToken  string `json:"access_token" binding:"required"`
	RefreshToken string `json:"refresh_token" binding:"required"`
}

// AuthResponse is returned on successful login or registration.
type AuthResponse struct {
	User         UserResponse `json:"user"`
	AccessToken  string       `json:"access_token"`
	RefreshToken string       `json:"refresh_token"`
}

type RefreshResponse struct {
	AccessToken  string `json:"access_token"`
	RefreshToken string `json:"refresh_token"`
}

// Forgot Password Flow
type ForgotPasswordRequest struct {
	Email string `json:"email" binding:"required,email"`
}

type VerifyResetPasswordOTPRequest struct {
	Email string `json:"email" binding:"required,email"`
	OTP   string `json:"otp" binding:"required,len=6"`
}

type VerifyResetPasswordOTPResponse struct {
	ResetToken string `json:"reset_token"`
}

type ResetPasswordRequest struct {
	ResetToken  string `json:"reset_token" binding:"required"`
	NewPassword string `json:"new_password" binding:"required,min=6"`
}

File: frontend/src/dtos/auth-dto.ts
// src/dtos/auth-dto.ts
import type { User } from '../models/user';


export interface LoginRequestDto {
  identifier: string;
  password: string;
}

export interface LoginResponseDto {
  user: User;
  accessToken: string;
  refreshToken: string;
}


export interface SendVerificationRequestDto {
  email: string;
}


export interface VerifyEmailRequestDto {
  email: string;
  otp: string;
}

export interface VerifyEmailResponseDto {
  verificationToken: string;
}


export interface CompleteRegistrationRequestDto {
  verificationToken: string;
  fullName: string;
  password: string;
}

export interface RegisterResponseDto {
  user: User;
  accessToken: string;
  refreshToken: string;
}


export interface RefreshTokenRequestDto {
  refreshToken: string;
}

export interface RefreshTokenResponseDto {
  accessToken: string;
  refreshToken: string;
}


export interface LogoutRequestDto {
  accessToken: string;
  refreshToken: string;
}



export interface ForgotPasswordRequestDto {
  email: string;
}

export interface VerifyResetOtpRequestDto {
  email: string;
  otp: string;
}

export interface VerifyResetOtpResponseDto {
  resetToken: string;
}

export interface ResetPasswordRequestDto {
  resetToken: string;
  newPassword: string;
}


export interface CompleteGoogleSetupRequestDto {
  setupToken: string;
  fullName: string;
}
